Client/Frontend/Autofill/CreditCard/CreditCardBottomSheet/SingleCreditCardView.swift

```diff
@@ -0,0 +1,306 @@
+            DispatchQueue.main.async { [weak self] in
+                self?.dismissVC()
+            }
+        }
+    }
+
+    @objc
+    private func didTapNotNow() {
+        dismissVC()
+    }
+
+    @objc
+    private func didTapManageCards() {
+        dismissVC()
+    }
+
+    // MARK: BottomSheet Delegate
+    func willDismiss() {
+    }
+}
+
+// MARK: UITableViewDelegate
+extension SingleCreditCardViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        return creditCardCell(indexPath: indexPath)
+    }
+
+    private func creditCardCell(indexPath: IndexPath) -> UITableViewCell {
+        guard let hostingCell = cardTableView.dequeueReusableCell(
+            withIdentifier: HostingTableViewCell<CreditCardItemRow>.cellIdentifier) as? HostingTableViewCell<CreditCardItemRow> else {
+            return UITableViewCell(style: .default, reuseIdentifier: "ClientCell")
+        }
+
+        let creditCard = viewModel.creditCard
+        let creditCardRow = CreditCardItemRow(
+            item: creditCard,
+            isAccessibilityCategory: UIApplication.shared.preferredContentSizeCategory.isAccessibilityCategory)
+        hostingCell.host(creditCardRow, parentController: self)
+        hostingCell.isAccessibilityElement = true
+        return hostingCell
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let headerView = tableView.dequeueReusableHeaderFooterView(withIdentifier: SingleCreditCardHeaderView.cellIdentifier) as? SingleCreditCardHeaderView else { return nil }
+        headerView.applyTheme(theme: themeManager.currentTheme)
+        headerView.viewModel = viewModel
+        return headerView
+    }
+
+    // easier to handle the bottom table spacing as a footer
+    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        switch viewModel.state {
+        case .save:
+            let emptyView = UIView()
+            return emptyView
+        case .update:
+            guard let footerView = tableView.dequeueReusableHeaderFooterView(withIdentifier: SingleCreditCardFooterView.cellIdentifier) as? SingleCreditCardFooterView else { return nil }
+            footerView.applyTheme(theme: themeManager.currentTheme)
+            if !footerView.manageCardsButton.responds(to: #selector(SingleCreditCardViewController.didTapManageCards)) {
+                footerView.manageCardsButton.addTarget(self, action: #selector(SingleCreditCardViewController.didTapManageCards), for: .touchUpInside)
+            }
+
+            return footerView
+        }
+    }
+
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return UX.distanceBetweenButtonAndTable
```

Comment:
The footer shouldn't have a fixed height - especially since a button can be in the footer which can grow with dynamic type.